lib/kubernetes-deploy/kubernetes_resource/replica_set.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class ReplicaSet < KubernetesResource
+    TIMEOUT = 5.minutes
+
+    def initialize(name:, namespace:, context:, file:, parent: nil, logger:)
+      @name = name
+      @namespace = namespace
+      @context = context
+      @file = file
+      @parent = parent
+      @logger = logger
+      @pods = []
+      @rollout_data = { "replicas" => 0 }
+    end
+
+    def sync
+      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
+
+      if @found = st.success?
+        rs_data = JSON.parse(raw_json)
+        interpret_json_data(rs_data)
+      else # reset
+        @rollout_data = { "replicas" => 0 }
+        @status = nil
+        @pods = []
+      end
+    end
+
+    def interpret_json_data(rs_data)
+      pods_data = get_pods(rs_data)
```

Comment:
I might be argued out of this one, but same as in the RS class, this should probably just return the objects.